### PR TITLE
add Array convert for data frame and data frame row

### PR DESCRIPTION
**Pull request by bkamins**
Following the discussion on Slack

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
Looks good to me

**Comment by ChrisRackauckas** [comment]
Thanks for this!

File: src/dataframerow/dataframerow.jl [hunk 1]
```diff
@@ -189,6 +189,11 @@ Base.convert(::Type{Vector{T}}, dfr::DataFrameRow) where T =
 Base.Vector(dfr::DataFrameRow) = convert(Vector, dfr)
 Base.Vector{T}(dfr::DataFrameRow) where T = convert(Vector{T}, dfr)
 
+Base.convert(::Type{Array}, dfr::DataFrameRow) = Vector(dfr)
```
Comment (nalimilan) [inline src/dataframerow/dataframerow.jl]: This is a bit more controversial as `DataFrameRow` could also be viewed as a single-row matrix. Maybe let's leave it out for now?

BTW, we define `Base.size(r::DataFrameRow) = (length(index(r)),)`, but isn't this at odds with how broadcasting works for `DataFrameRow`? I.e. it could also be `(1,length(index(r)))`.

Comment (bkamins) [reply src/dataframerow/dataframerow.jl]: We allow `Vector` to be called on  `DataFrameRow`. Since we do this then all other interpretations are ruled out I think. Doing `df[row, cols]` drops one level of dimensionality from a data frame to make it 1-dimensional object just like in arrays.

Note that we do not allow:
```
julia> using DataFrames

julia> df = DataFrame(ones(2,3))
2×3 DataFrame
│ Row │ x1      │ x2      │ x3      │
│     │ Float64 │ Float64 │ Float64 │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1.0     │ 1.0     │ 1.0     │
│ 2   │ 1.0     │ 1.0     │ 1.0     │

julia> df[1, :] .= [1, 2, 3]
DataFrameRow
│ Row │ x1      │ x2      │ x3      │
│     │ Float64 │ Float64 │ Float64 │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1.0     │ 2.0     │ 3.0     │
```
but we do not allow:
```
julia> df[1, :] .= [1 2 3]
ERROR: DimensionMismatch("cannot broadcast array to have fewer dimensions")
```
but we allow (as `1:1` does not drop a dimension):
```
julia> df[1:1, :] .= [1 2 3]
1×3 SubDataFrame
│ Row │ x1      │ x2      │ x3      │
│     │ Float64 │ Float64 │ Float64 │
├─────┼─────────┼─────────┼─────────┤
│ 1   │ 1.0     │ 2.0     │ 3.0     │
```

Currently we do not allow:
```
julia> df[1, :] .= df[1, :]
ERROR: ArgumentError: broadcasting over `DataFrameRow`s is reserved
```
(i.e. for `DataFrameRow` to be on RHS of broadcasted assignment, as we are waiting for Base to decide how to treat `NamedTuple` in the same place).

In summary: all the infrastructure of DataFrames.jl for `DataFrameRow` assumes it is 1-dimensional.

Comment (nalimilan) [reply src/dataframerow/dataframerow.jl]: OK, I guess that's fine then.

**Comment by bkamins** [comment]
Thank you!

**Comment by ChrisRackauckas** [comment]
Awesome! Any idea on when this will get released?

**Comment by bkamins** [comment]
I think in a few days at most. See https://github.com/JuliaData/DataFrames.jl/pull/2028 for the list of pending PRs (essentially 1 PR that is finished but requires a final check + possibly 2 PRs that we will exclude from that release if we cannot get consensus on them)
